packages/patterns/src/components/create-pattern-modal.js

```diff
@@ -42,6 +42,44 @@ export default function CreatePatternModal( {
 	const { saveEntityRecord, invalidateResolution } = useDispatch( coreStore );
 	const { createErrorNotice } = useDispatch( noticesStore );
 
+	const { corePatternCategories, userPatternCategories } = useSelect(
+		( select ) => {
+			const { getUserPatternCategories, getBlockPatternCategories } =
+				select( coreStore );
+
+			return {
+				corePatternCategories: getBlockPatternCategories(),
+				userPatternCategories: getUserPatternCategories(),
+			};
+		}
+	);
+
+	const categoryMap = useMemo( () => {
+		// Merge the user and core pattern categories and remove any duplicates.
+		const categories = [
+			...userPatternCategories,
+			...corePatternCategories,
+		].reduce( ( uniqueCategories, category ) => {
+			if (
+				! uniqueCategories.has( category.label ) &&
+				// There are two core categories with `Post` label so explictily remove the one with
+				// the `query` slug to avoid any confusion.
+				category.name !== 'query'
+			) {
+				// We need to store the name separately as this is used as the slug in the
+				// taxonomy and may vary from the label.
+				uniqueCategories.set( category.label, {
+					label: category.label,
+					value: category.label,
+					name: category.name,
+				} );
+			}
+			return uniqueCategories;
+		}, new Map() );
```

Comment:
Nit, non-blocking: Personally I would think a `for` loop could be easier to understand, but I can read this just fine too 👍 .

Reply:
Most people reading the code are probably still more use to simple `for` loops but I could be biased given that's my preference given too many use generic variable names in `reduce` e.g. `accumulator`. That's obviously not the case here though.

Reply:
I quite like the slightly more self-contained nature of the .reduce for this sort of thing, but for the sake of quick readability let's switch it to a forEach - done.